stl/inc/xmemory

```diff
@@ -853,6 +853,10 @@ public:
         return static_cast<size_t>(-1) / sizeof(_Ty);
     }
 #endif // _HAS_DEPRECATED_ALLOCATOR_MEMBERS
+
+#ifdef __SANITIZE_ADDRESS__
+    static constexpr size_t _Minimum_allocation_alignment = 8;

```

Comment:
What is this?

Reply:
Casey asked me to add that into `std::allocator` instead of special casing it.

Reply:
I don't get what happens  with non-default allocator

Reply:
Ah I see. If an allocator provides a `_Minimum_allocation_alignment >= 8` we don't insert the run-time checks to determine the next 8-byte aligned address. We also check if the allocator will return sufficiently aligned addresses by looking at the alignment of the type it is meant for if `_Minimum_allocation_alignment` is not present. Essentially it's a way to tell all the annotation code that no run-time alignment needs to be done.

Reply:
I see, thanks